src/SessionManager/ProgramSettings.cc

```diff
@@ -34,6 +37,89 @@ void applyOptionalArgument(T& val, const string& argument_name, const cxxopts::P
 }
 
 ProgramSettings::ProgramSettings(int argc, char** argv) {
+    if (argc > 1) {
+        spdlog::info("Using command-line settings");
```

Comment:
I think this and the other two messages should be `debug`.

Reply:
I attempted to change this to debug, but it won't print since the Logger is not yet initialized at that point. I would suggest removing that message line. It's quite obvious if you are starting the backend with or without command line arguments unless you are using an alias or something else. What do you think? Maybe @veggiesaurus wants to comment too.

This is how it usually looks like:
```
root@bc0736f1a5a3:~/carta-backend/build# ./carta_backend --verbosity 5
[2021-04-19 09:38:36.589] [info] Using command-line settings
[2021-04-19 09:38:36.593] [info] Writing to the log file: /root/.carta/log/carta.log
[2021-04-19 09:38:36.593] [info] /root/carta-backend/build/carta_backend: Version 2.0.0-dev.21.04.13
[2021-04-19 09:38:36.601] [warning] Could not find CARTA frontend files in directory /root/carta-backend/build/../share/carta/frontend.
[2021-04-19 09:38:36.601] [warning] Failed to host the CARTA frontend. Please specify a custom location using the frontend_folder argument.
[2021-04-19 09:38:36.604] [info] Listening on port 3002 with top level folder /, starting folder /root/carta-backend/build. The number of OpenMP worker threads will be handled automatically.
^C[2021-04-19 09:38:37.963] [info] Exiting backend.
```

And here's proof that it won't appear if you set it to `spdlog::debug`, even if asking for verbosity level 5 (debug):
```
root@bc0736f1a5a3:~/carta-backend/build# ./carta_backend --verbosity 5
[2021-04-19 09:34:01.189] [info] Writing to the log file: /root/.carta/log/carta.log
[2021-04-19 09:34:01.189] [info] /root/carta-backend/build/carta_backend: Version 2.0.0-dev.21.04.13
[2021-04-19 09:34:01.191] [warning] Could not find CARTA frontend files in directory /root/carta-backend/build/../share/carta/frontend.
[2021-04-19 09:34:01.191] [warning] Failed to host the CARTA frontend. Please specify a custom location using the frontend_folder argument.
[2021-04-19 09:34:01.191] [info] Listening on port 3002 with top level folder /, starting folder /root/carta-backend/build. The number of OpenMP worker threads will be handled automatically.
^C[2021-04-19 09:34:02.219] [info] Exiting backend.
```

Reply:
Why not defer the printing until after the logger is initialised in `Main.cc`? You could save a message summarising the source(s) of the configuration in its own field on the settings object, or pass a separate output parameter when calling `ProgramSettings` -- but that's a bit C-ish and may be kind of gross to do in a constructor, so maybe not. I think a single summarised message (at debug level) would be fine.

Reply:
I don't see why we need to print this line in the first place. If you put command-line options in and they are incorrect, we'll get an error. If you put correct ones in, I'm pretty sure you'd expect them to be used.

Reply:
The other issue here is that this message will _always_ go to stdout, because the log hasn't been initialised. So your actual log file will be missing this line anyway.

Reply:
The message about the commandline options is the least relevant, but the message(s) about the locations of the config files could be useful (at debug level).

Those messages should either be logged at debug level after the logger is initialised or not printed at all -- I think this is too verbose for info.